tests/lean/run/decideTactic.lean

```diff
@@ -0,0 +1,102 @@
+/-!
+# Tests of the `decide` tactic
+-/
+
+/-!
+Success
+-/
+#guard_msgs in
+example : 2 + 2 ≠ 5 := by decide
+
+
+/-!
+False proposition
+-/
+
+/--
+error: tactic 'decide' proved that the proposition
+  1 ≠ 1
+is false
+-/
+#guard_msgs in
+example : 1 ≠ 1 := by decide
+
+
+/-!
+Irreducible decidable instance
+-/
+opaque unknownProp : Prop
+
+/--
+error: tactic 'decide' failed for proposition
+  unknownProp
+since its Decidable instance
+  Classical.propDecidable unknownProp
+did not reduce to isTrue or isFalse.
+
+After unfolding the instance Classical.propDecidable, reduction got stuck at the Decidable instance
+  Classical.choice ⋯
+
+Hint: Reduction got stuck on Classical.choice, suggesting that there are "classical" instances, due
+to for example an 'open scoped Classical' command. The 'decide' tactic works by evaluating a
+decision procedure via reduction, but such classical instances cannot be evaluated.
```

Comment:
I think this message might be a bit hard to figure out for non-experts, and while the double-quotes make sense, I worry that they split the audiences a bit. Here's a suggestion:

```suggestion
Hint: Reduction got stuck on 'Classical.choice', which indicates that the 'Decidable' instance
depends on classical reasoning. Classical reasoning principles cannot be run as programs, and
the 'decide' tactic works by evaluating a decision procedure via reduction, which then gets
stuck. Instances that depend on classical reasoning are most often in scope due to an
'open scoped Classical' command.
```

Reply:
I've got another iteration that tries to explain exactly what "classical" means in this context. I also toned down "most often in scope" since in mathlib it's probably about as common to use the `classical` tactic.

Reply:
This is really good now!